### PR TITLE
Added Breadth First Tree Traversal

### PR DESCRIPTION
**Pull request by Dalfonso06**
Breadth First Tree traversal uses the breadth first algorithm to traverse through binary trees which can be used for algorithms such as level-order traversal and finding the deepest node of a binary tree with a time complexity of O(n) and space complexity of O(w) where w is the maximum width of the tree.
 
- [X] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Comments in areas I changed are up to date
- [ ] I have added comments to hard-to-understand areas of my code
- [X] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Algorithms.csproj [hunk 1]
```diff
@@ -28,4 +28,10 @@
     <ProjectReference Include="..\Utilities\Utilities.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Remove="Graph\Dijkstra\" />
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Graph\Dijkstra\" />
+  </ItemGroup>
```
Comment (siriak) [inline Algorithms/Algorithms.csproj:36]: Please remove this

File: Algorithms/Algorithms.sln [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{C2FEFDE9-796C-46D7-8C98-5CC83034083D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{C2FEFDE9-796C-46D7-8C98-5CC83034083D}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{C2FEFDE9-796C-46D7-8C98-5CC83034083D}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{C2FEFDE9-796C-46D7-8C98-5CC83034083D}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
```
Comment (siriak) [inline Algorithms/Algorithms.sln:22]: Please remove this file

**Review by siriak — CHANGES_REQUESTED** [review]

File: Algorithms/Algorithms.sln (hunk 2 above)
Comment (Dalfonso06) [reply Algorithms/Algorithms.sln]: Just removed it, not really sure why that was added

File: Algorithms/Algorithms.csproj [hunk 3]
```diff
@@ -27,5 +27,4 @@
     <ProjectReference Include="..\DataStructures\DataStructures.csproj" />
     <ProjectReference Include="..\Utilities\Utilities.csproj" />
   </ItemGroup>
```
Comment (siriak) [inline Algorithms/Algorithms.csproj:29]:
```suggestion
  </ItemGroup>

```

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**